### PR TITLE
#169231229 Configure data flow

### PR DESCRIPTION
**Pull request by hezzie**
#### What does this PR do?
This PR configures the flow of data
#### Description of Task to be completed?
- Consume the open weather API
- Create a Form component.
- Create functions to handle the transfer of data from Form(child) component to App(parent).
#### How should this be manually tested?
- Clone this repo
- cd . into ch-data-flow-169231229 branch
- install the dependencies (npm install)
- Start the app (npm, start)
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169231229](https://www.pivotaltracker.com/n/projects/2406895/stories/169231229)
#### Screenshot?
N/A

### CODE REVIEW

**Comment by NiyoEric** [comment]
I have 2 feedback for you
 1. Hide API **APP_KEY** in your **.env** and use process.env to access it.
 2. Can you display the city name as our form placeholder from the state so that users understand why by default we display Kigali?

File: src/App.js [hunk 1]
```diff
@@ -1,8 +1,61 @@
...
 
-const App = () => {
-  return <h1>Hello world</h1>;
+dotenv.config();
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      city: "kigali",
```
Comment (kipropbrian) [inline src/App.js]: This should not be hard coded

File: src/App.js [hunk 2]
```diff
@@ -1,8 +1,61 @@
...
+  handleSubmit = event => {
+    event.preventDefault();
+    this.findCity();
+  };
+
+  findCity = async () => {
+    try {
+      const { city } = this.state;
+      const { REACT_APP_API_KEY: Key } = process.env;
+      const url = `http://api.openweathermap.org/data/2.5/forecast?q=${city}&APPID=${Key}`;
```
Comment (kipropbrian) [inline src/App.js]: The URL should also not be hardcoded

File: src/components/Form.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from "react";
+
+class Form extends Component {
```
Comment (kipropbrian) [inline src/components/Form.js:3]: Is this a general form? If not the name should reflect what exactly it is doing.

**Review by kipropbrian — CHANGES_REQUESTED** [review]
Left some change requests and questions. Also
The PR description format is not looking very good. Check here for how to style it.
https://github.com/andela/engineering-playbook/tree/c74e0b42578fcf45d76b91e43b85d5159821b7ea/5.%20Developing/Conventions

There should be a PR for the PR_template.md this will pre-format your PR's.

File: src/App.js (hunk 1 above)
Comment (hezzie) [reply src/App.js]: Okey Let me work on it.

File: src/App.js [hunk 4]
```diff
@@ -1,8 +1,94 @@
...
+  render() {
+
+
+    return (
+      <div>
+        <Form
+          updateCity={this.updateCity}
+          handleSubmit={this.handleSubmit}
+          handleChange={this.handleChange}
+          state={this.state}
```
Comment (kiproping) [inline src/App.js]: Since you just require the city, why not `city={this.state.city}`

Comment (kipropbrian) [inline src/App.js]: Since you just need the city from the state, just do `city={this.state.city}`

**Review by victorkarangwa — COMMENTED** [review]
Good to go!

**Review by WillyBoris — APPROVED** [review]

**Review by NiyoEric — APPROVED** [review]

**Review by victorkarangwa — APPROVED** [review]

**Review by WillyBoris — APPROVED** [review]

File: src/components/Form.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  render() {
+    const city = this.props.city;
```
Comment (kiproping) [inline src/components/Form.js]: There is no props called city.

**Review by WillyBoris — APPROVED** [review]

**Pull request merged**